numpy/linalg/linalg.py

```diff
@@ -2112,6 +2112,9 @@ def norm(x, ord=None, axis=None, keepdims=False):
     """
     x = asarray(x)
 
+    if not issubclass(x.dtype.type, floating):
```

Comment:
One more thing. Did you mean either `float` or `numpy.float`? This is the new point of failure in the tests.